plotly/tools.py

```diff
@@ -298,15 +297,21 @@ def get_embed(file_owner_or_url, file_id=None, width="100%", height=525):
                 "'{1}'."
                 "\nRun help on this function for more information."
                 "".format(url, plotly_rest_url))
-        head = plotly_rest_url + "/~"
-        file_owner = url.replace(head, "").split('/')[0]
-        file_id = url.replace(head, "").split('/')[1]
+        urlsplit = six.moves.urllib.parse.urlparse(url)
+        file_owner = urlsplit.path.split('/')[1].split('~')[1]
+        file_id = urlsplit.path.split('/')[2]
+
+        # to check for share_key we check urlsplit.query
+        query_dict = six.moves.urllib.parse.parse_qs(urlsplit.query)
+        if query_dict:
+            share_key = query_dict['share_key'][-1]
+        else:
+            share_key = ''
     else:
         file_owner = file_owner_or_url
-    resource = "/apigetfile/{file_owner}/{file_id}".format(file_owner=file_owner,
-                                                           file_id=file_id)
+        share_key = ''
     try:
-        test_if_int = int(file_id)
+        test_if_int = int(0 if file_id is None else file_id)
```

Comment:
? this will break just below if `file_id=None`, we use `int(file_id)` which can raise an Error?

Why would we want `file_id` to default to `0` anyways?

Did you change this because `None` was raising a `TypeError` instead of a `ValueError`? If that's the case, we should just:

`except (ValueError, TypeError):`